include/alpaka/dev/Traits.hpp

```diff
@@ -64,6 +64,20 @@ namespace alpaka
                 typename T,
                 typename TSfinae = void>
             struct Reset;
+
+            //#############################################################################
+            //! The device register queue trait.
+            template<
+                typename T,
+                typename TSfinae = void>
+            struct RegisterQueue;
+
+            //#############################################################################
+            //! The device get all queues trait.
+            template<
+                typename T,
+                typename TSfinae = void>
+            struct GetAllQueues;
```

Comment:
Do we need those public traits at all?
The `ICpuQueue` interface with the `registerQueue` and `getAllQueues` methods is an implementation detail of the CPU queues (and therefore lives in the `cpu` subfolder) and should not be publicly visible even after your changes. The CUDA queues do not implement this and do not need this.
It should be possible to live without those traits because the `QueueCpuGenericXxx` gets its specific device implementation as template parameter and the device simply has to implement a `RegisterQueue` method without the need for any explicit public interface.

Reply:
So, it would be better to require TDev to implement some concept `QueueRegister`. I agree with skipping the traits.

Reply:
Just one issue: DevCpu does not have `GetAllQueues()` and `RegisterQueue()`, they are only present in DevCpuImpl. In introduced the traits because it looked a it strange when the queue would just reach into the internal implementation of DevCpu. So how should this be solved?
1. Just access `dev.m_spDevOmp4Impl->RegisterQueue(queue.m_spQueueImpl)`.
2. Move the member functions to DevCpu, as they are part of its interface.

Reply:
both sound reasonable so do what you think is easier. For the first solution you might get access issues and the members have to have the same name so the second might be easier.

Reply:
I will do option 2 then, because it is cleaner, though  more work than 1.